### ISSUE TITLE
Save configuration state in text file rather than Marshalled data

### ISSUE THREAD

**Issue by rixed** [rixed/ramen#140]
1. Makes it editable/auditable by hand
2. Avoid blaming Marshal for all segfault

**Comment by rixed** [comment]
PPP v2.0 now allow several parsers for a single type.

**Comment by rixed** [comment]
3. independant of ocaml version

**Comment by rixed** [comment]
Might require to split the program into 1 file per operation to speed up serialization and have finer grained locks. Could be stored in directories named after the program and operation would be files (as the only forbidden char in operation name is the '/' - and we could add the '\0').

**Comment by rixed** [comment]
Depends on #146

**Comment by rixed** [comment]
Tried on program config files and it was huge, slow, and of little value since the "text" is mostly a verbose, very deep AST that's very unpractical to read, let alone to edit.
There is still value in having the "sql" of a program editable and then automatically recompiled whenever is changes, but that's another story.
